Source/IntegrationTests/TestFiles/LitTests/LitTest/comp/replaceables/complex/user.dfy

```diff
@@ -0,0 +1,17 @@
+// RUN: %baredafny run %args %s --target=java --build=%S/Build/build --input %S/Inputs/wrappers.dfy --input %S/Inputs/random.dfy --input %S/Inputs/randomJava.dfy --input %S/Inputs/Interop/__default.java > "%t"
+// RUN: %baredafny run %args %s --build=%S/Build/build --input %S/Inputs/wrappers.dfy --input %S/Inputs/random.dfy --input %S/Inputs/randomCSharp.dfy --input %S/Inputs/Interop.cs >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+// Test can fail non-deterministically by design, but the chance is ~10^-30
+method Main() {
+  var x := DfyRandom.GetRandomNat(1000000);
+  var y := 0;
+  var i := 5;
+  while(x != y && i != 0)
+    decreases i
+  {
+    y := DfyRandom.GetRandomNat(1000000);
+    i := i - 1;
+  }
+  expect x != y;
```

Comment:
```suggestion
  expect i == 0 || x != y;
```
I hope you don't mind too much but I'd feel better for maintenance if the expect was always true :-)

Reply:
:-( but then it doesn't test the randomness. Anyways, made the change.

Reply:
hmmm. Ok. In that case, can you mark Main as `decreases *` and that way you can also mark the loop as `decreases *`. Then you are truly testing for randomness, but now it's no longer a probability on the test failing or not, it's how long it will take.